packages/components/src/Autocomplete/Menu.tsx

```diff
@@ -63,7 +76,16 @@ export function Menu({
             key={option.value}
             onMouseDown={onOptionSelect.bind(undefined, option)}
           >
-            <div className={styles.icon}>
+            <div
+              className={styles.icon}
+              ref={ref => {
+                if (!menuItemRefs.current[index]) {
+                  const newMenuRef = ref as HTMLDivElement;
+                  menuItemRefs.current[index] = newMenuRef;
+                }
+                return menuItemRefs.current[index];
+              }}
```

Comment:
This looks really similar to the above code, too. Maybe it should be refactored into a method?

Reply:
Good idea, refactored into it's own method!